### PR TITLE
 [Idea]: DLX implementation #1516 

### LINKED ISSUE #1516: [Idea]: DLX implementation
### Idea Title

Dancing Links (DLX) (Exact Cover Problems) implementation 

### Idea Description

Add this algorithm  to have more to learn.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

Assign me this under hacktoberfest and gssoc-ext

### PR DESCRIPTION
**Pull request by AE-Hertz**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

 [Idea]: DLX implementation #1516 

Fixes #1516 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/DLX algorithm/DLX.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
+---
+
+id: Dancing-Links
```
Comment (ajay-dhangar) [inline docs/algorithms/DLX algorithm/DLX.md]: replace `id: Dancing-Links` with `id: dancing-links`

File: docs/algorithms/DLX algorithm/DLX.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+### Time Complexity
+
+- **Average Case**: The time complexity depends on the size of the matrix and the specific problem being solved but is generally more efficient than brute force methods.
+- **Worst Case**: The worst-case complexity can be exponential in the number of rows and columns, but the algorithm is highly optimized for sparse matrices.
+
+### Space Complexity
+
+- **Space Complexity**: \( O(n + m) \), where \( n \) is the number of rows and \( m \) is the number of columns in the matrix. The space is primarily used to store the linked list nodes for the matrix representation.
+
```
Comment (ajay-dhangar) [inline docs/algorithms/DLX algorithm/DLX.md]: use `$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by AE-Hertz** [comment]
> update your code based on suggestions

resolved

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**